analyze/benchcases.py

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env python
+
+from enum import Enum
+
+__all__ = ['BenchmarkCase', 'cases']
+
+
+class BenchmarkCase(Enum):
+    CLIENT2TEE = 'client2tee'
+    KMOD2TEE = 'kernel2tee'
+
+    def __str__(self):
+        return str(self.value)
```

Comment:
There is no `self.value` in the class by default(?). I.e., you should probably initialize self.value in the constructor.

Reply:
Enum is not a regular python class, and here is a magic explained :) :

https://docs.python.org/3/library/enum.html#creating-an-enum

> class Color(Enum):
...     RED = 1
...     GREEN = 2
...     BLUE = 3
The class Color is an enumeration (or enum)
The attributes Color.RED, Color.GREEN, etc., are enumeration members (or enum members) and are functionally constants.
The enum members have names and values (the name of Color.RED is RED, the value of Color.BLUE is 3, etc.)
Even though we use the class syntax to create Enums, Enums are not normal Python classes (Enums have a custom metaclass that affects many aspects of both derived Enum classes and their instances (members).)

So in this case `self` will point to the member here, and enum members has preinitialized `.value` and `.name` attributes.

to understand the whole picture, just an example (some hypothetical example with `Planet` enumeration ):
```
>>> dir(Planet)
['EARTH', 'JUPITER', 'MARS', 'MERCURY', 'NEPTUNE', 'SATURN', 'URANUS', 'VENUS', '__class__', '__doc__', '__members__', '__module__']
>>> dir(Planet.EARTH)
['__class__', '__doc__', '__module__', 'name', 'surface_gravity', 'value']
```

Reply:
Ah, I see and I didn't pay attention that the class inherited Enum, thanks for the explanation!